### PR TITLE
Added a retrieve endpoint

### PR DESCRIPTION
**Pull request by emmberk**
You can can retrieve products from S3 with a new /retrieve/<ingestId> endpoint. The product should download as its original filename+extension. This works in the browser.

### CODE REVIEW

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
OWASP failed because of a known issue downloading CVEs.

File: ingest/src/main/java/com/connexta/ingest/endpoint/IngestController.java [hunk 1]
```diff
@@ -38,12 +42,20 @@ public IngestController(IngestService ingestService) {
...
+    } catch (IOException e) {
+      LOGGER.warn("Unable to ingest", e);
+      return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
     }
-    return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
-    // TODO: Send out the Transformation request to process the new data
+  }
+
+  @GetMapping("/retrieve/{ingestId}")
+  public ResponseEntity<Resource> retrieve(@PathVariable final String ingestId) {
```
Comment (ahoffer) [inline ingest/src/main/java/com/connexta/ingest/endpoint/IngestController.java]: :question: Does the caller need the ResponseEntity or just the Resource?

File: ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java [hunk 2]
```diff
@@ -9,30 +9,60 @@
...
       MultipartFile file,
       String title,
-      String fileName) {
+      String fileName)
+      throws IOException {
+    final String ingestId = UUID.randomUUID().toString().replace("-", "");
+    s3Adaptor.upload(
+        new IngestRequest(acceptVersion, fileSize, mimeType, file, title, fileName), ingestId);
+
+    final String url = new URL("TODO/retrieve/" + ingestId).toString();
```
Comment (Kjames5269) [inline ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java]: ✏️ This URL should come from a configuration file

Comment (emmberk) [reply ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java]: Yep. This is a know issue. We will address this in a future PR.

File: ingest/src/main/java/com/connexta/ingest/adaptors/S3Adaptor.java [hunk 3]
```diff
@@ -62,25 +68,38 @@ private void initializeS3() {
...
+            .key(key)
+            .contentType(ingestRequest.getMimeType())
+            .contentLength(ingestRequest.getFileSize())
+            .metadata(ImmutableMap.of("filename", fileName))
+            .build(),
+        RequestBody.fromInputStream(
+            ingestRequest.getFile().getInputStream(), ingestRequest.getFile().getSize()));
+  }
+
+  public ResponseEntity<Resource> retrieve(String ingestId) {
```
[the remark this answers is not in the document]
Comment (ahoffer) [reply ingest/src/main/java/com/connexta/ingest/adaptors/S3Adaptor.java]: I was thinking of signature line line:
`  public Resource retrieve(String ingestId)

**Comment by emmberk** [comment]
I manually tested this. `./gradlew clean build` and `./gradlew dependencyCheckAnalyze --info` succeeded for me.

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
The CI build failed because of a known issue downloading CVEs. I am overriding the CI build because OWASP was tested locally.

**Pull request merged**